crates/sui-indexer-alt-e2e-tests/src/lib.rs

```diff
@@ -298,6 +269,44 @@ impl FullCluster {
     }
 }
 
+pub struct OffchainClusterConfig {
+    pub indexer_args: IndexerArgs,
+    pub consistent_indexer_args: IndexerArgs,
+    pub client_args: ClientArgs,
+    pub indexer_config: IndexerConfig,
+    pub consistent_config: ConsistentConfig,
+    pub jsonrpc_config: JsonRpcConfig,
+    pub graphql_config: GraphQlConfig,
+    pub registry: prometheus::Registry,
+    pub cancel: CancellationToken,
+}
+
+impl OffchainClusterConfig {
+    pub fn with_local_ingestion() -> Self {
+        let temp_dir = tempfile::tempdir()
+            .context("Failed to create data ingestion path")
+            .unwrap();
+        let client_args = ClientArgs {
+            local_ingestion_path: Some(temp_dir.path().to_owned()),
+            remote_store_url: None,
+            rpc_api_url: None,
+            rpc_username: None,
+            rpc_password: None,
+        };
```

Comment:
I'm not sure this will work reliably -- `tempdir` returns a guard object -- `TempDir` -- that keeps the directory alive until it is dropped. Here there is nothing to keep it alive, so I would expect in some cases the test will fail unexpectedly because the temporary directory was cleaned up too quickly.

This problem will go away if you remove `ClientArgs` from the config and make it the responsibility of e.g. the `FullCluster` or whatever to set it up, and then hold on to the temp directory.